util/ranger/detacher.go

```diff
@@ -446,6 +446,34 @@ func allEqOrIn(expr expression.Expression) bool {
 	return false
 }
 
+// accessCondShouldReserve checks whether access condition should be reserved to filter condition.
+// NOTE: access must be EQ/IN function or LogicOr of EQ/IN functions
+func accessCondShouldReserve(access expression.Expression, length int) bool {
+	f, _ := access.(*expression.ScalarFunction)
+	switch f.FuncName.L {
+	case ast.LogicOr:
+		return accessCondShouldReserve(f.GetArgs()[0], length) || accessCondShouldReserve(f.GetArgs()[0], length)
```

Comment:
Hello, I didn't get the recursion logic here, why we only evaluate `f.GetArgs()[0]`?

Reply:
Sorry, I made a mistake. The second should be `f.GetArgs()[1]`.